lib/realtime/repo.ex

```diff
@@ -52,41 +52,39 @@ defmodule Realtime.Repo do
   end
 
   @doc """
-  Converts a Postgrex.Result into a given struct
+  Deletes all records for a given query and converts them into a given struct
   """
-  @spec result_to_single_struct({:ok, Postgrex.Result.t()} | {:error, any()}, module()) ::
-          {:ok, struct()} | {:ok, nil} | {:error, any()}
-  def result_to_single_struct({:ok, %Postgrex.Result{rows: [row], columns: columns}}, struct) do
+  @spec del(DBConnection.conn(), Ecto.Queryable.t()) ::
+          {:ok, non_neg_integer()} | {:error, any()}
+  def del(conn, query) do
+    with {:ok, %Postgrex.Result{num_rows: rows}} <- run_delete_query(conn, query) do
+      {:ok, rows}
+    end
+  rescue
+    error -> {:error, error}
```

Comment:
Below we have the following test:

```elixir
   test "handles error on wrong delete", %{conn: conn} do
       # wrong id type
       query = from c in Channel, where: c.id == "potato"
       assert {:error, error} = Repo.del(conn, query)
       assert is_struct(error, Ecto.QueryError)
```

I'd carefully consider using `rescue _ ->` here. It may catch errors that arguably shouldn't. In this particular case I'd say the query is _wrong_ and so it's a programmers bug and del/1 should raise. In other words the fact it's invalid data should have been caught sooner by for example an `Integer.parse(id)` check. If this doesn't sound practical, I'd consider a narrower handling e.g. `rescue Ecto.QueryError -> `

What do you think?

Reply:
I think you are right! will revert to not catch the exception as it could be hiding some really bad interactions with such a destructive operation 👍